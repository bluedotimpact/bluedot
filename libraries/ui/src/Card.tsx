import React from 'react';
import { CTALinkOrButton } from './CTALinkOrButton';

interface CardProps {
  imageSrc: string;
  title: string;
  subtitle?: string;
  ctaUrl?: string;
}

export const Card: React.FC<CardProps> = ({
  imageSrc,
  title,
  subtitle,
  ctaUrl,
}) => {
  return (
    <div className="card w-80 h-80 flex flex-col items-start">
      <div className="card__image-container w-full h-48 mb-8">
        <img
          className="card__image w-full h-full object-cover rounded-2xl"
          src={imageSrc}
          alt={`${title}`}
        />
      </div>

      <div className="card__content flex flex-col gap-8">
        <div className="card__text space-y-1">
          <h2 className="card__title text-2xl font-semibold text-bluedot-darker">{title}</h2>
          {subtitle && (<p className="card__subtitle text-base text-black">{subtitle}</p>)}
        </div>

        <CTALinkOrButton
          url={ctaUrl}
          variant="secondary"
          withChevron={false}
        >
          LinkedIn
        </CTALinkOrButton>
      </div>
    </div>
  );
};

export default Card;
