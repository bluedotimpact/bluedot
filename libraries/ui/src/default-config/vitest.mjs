/* eslint-disable import/no-extraneous-dependencies */
import { defineConfig, defaultExclude } from 'vitest/config';
import react from '@vitejs/plugin-react';
import * as dotenv from 'dotenv';

/**
 * @param {import('vitest/config').UserConfig} [config] 
 * @returns {import('vitest/config').UserConfig}
 */
export const withDefaultBlueDotVitestConfig = async (config) => defineConfig({
  ...config,
  plugins: [react(), ...(config?.plugins ?? [])],
  test: {
    environment: 'happy-dom',
    env: dotenv.config({ path: '.env.test' }).parsed,

    ...config?.test,

    exclude: [...defaultExclude, '**/.{turbo,next}/**', ...(config?.test?.exclude ?? [])],
  },
});
