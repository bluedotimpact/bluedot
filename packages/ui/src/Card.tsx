import clsx from 'clsx';
import { ButtonProps, LinkOrButton } from './Button';

export type CardProps = React.PropsWithChildren<{
  className?: string
}>;

export const Card: React.FC<CardProps> = ({ children, className }) => {
  return (
    <div className={clsx('border border-neutral-500 rounded px-8 pb-4', className)}>
      {children}
    </div>
  );
};

export const CardButton: React.FC<ButtonProps> = ({ className, ...rest }) => {
  return (
    <LinkOrButton className={clsx('border border-neutral-500 rounded px-8 pb-4 text-black transition-all duration-200 inline-block cursor-pointer data-[hovered]:border-bluedot-normal data-[hovered]:bg-bluedot-lighter data-[focus-visible]:border-bluedot-normal data-[focus-visible]:bg-bluedot-lighter data-[pressed]:border-bluedot-normal data-[pressed=true]:bg-bluedot-normal data-[pressed=true]:text-white outline-none [text-align:inherit]', className)} {...rest} />
  );
};
