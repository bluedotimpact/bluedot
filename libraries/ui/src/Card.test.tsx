import { describe, expect, test } from 'vitest';
import { render } from '@testing-library/react';
import { Card } from './Card';

describe('Card', () => {
  const defaultProps = {
    imageSrc: '/images/team/member.jpg',
    title: 'John Doe',
    subtitle: 'Developer',
    ctaUrl: 'https://linkedin.com/in/johndoe',
    ctaText: 'LinkedIn',
  };

  test('renders default as expected', () => {
    const { container } = render(<Card {...defaultProps} />);
    expect(container).toMatchSnapshot();

    const anchorElement = container.querySelector('a.card__cta');
    expect(anchorElement).not.toBeNull();
    expect(anchorElement?.getAttribute('href')).toBe(defaultProps.ctaUrl);
  });

  test('renders with isEntireCardClickable as expected', () => {
    const { container } = render(<Card {...defaultProps} isEntireCardClickable />);
    expect(container).toMatchSnapshot();

    const anchorElement = container.querySelector('a.card');
    expect(anchorElement).not.toBeNull();
    expect(anchorElement?.getAttribute('href')).toBe(defaultProps.ctaUrl);
  });

  test('renders with custom className', () => {
    const { container } = render(
      <Card
        {...defaultProps}
        className="custom-class"
      />,
    );
    const cardElement = container.querySelector('.custom-class');
    expect(cardElement).not.toBeNull();
  });

  test('does not include footer unless given', () => {
    const { container } = render(<Card {...defaultProps} />);
    const ctaMetadataElement = container.querySelector('.card__footer');
    expect(ctaMetadataElement).toBeNull();
  });

  test('includes footer when given', () => {
    const { container } = render(
      <Card {...defaultProps}>A footer of some kind</Card>,
    );
    const ctaMetadataElement = container.querySelector('.card__footer');
    expect(ctaMetadataElement).not.toBeNull();
  });
});
