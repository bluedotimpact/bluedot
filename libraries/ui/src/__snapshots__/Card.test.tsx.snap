// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Card > renders default as expected 1`] = `
<div>
  <div
    class="card w-80 h-80 flex flex-col items-start"
  >
    <div
      class="card__image-container w-full h-48 mb-8"
    >
      <img
        alt="John Doe"
        class="card__image w-full h-full object-cover rounded-2xl"
        src="/images/team/member.jpg"
      />
    </div>
    <div
      class="card__content flex flex-col gap-8"
    >
      <div
        class="card__text space-y-1"
      >
        <h2
          class="card__title text-2xl font-semibold text-bluedot-darker"
        >
          John Doe
        </h2>
        <p
          class="card__subtitle text-base text-black"
        >
          Developer
        </p>
      </div>
      <a
        class="cta-button flex items-center justify-center rounded-lg transition-all duration-200 font-semibold text-base cta-button--secondary bg-transparent text-bluedot-normal border border-bluedot-normal hover:bg-bluedot-lighter px-4 py-2"
        href="https://linkedin.com/in/johndoe"
        rel="noopener noreferrer"
        target="_blank"
      >
        <span
          class="cta-button__text"
        >
          LinkedIn
        </span>
      </a>
    </div>
  </div>
</div>
`;

exports[`Card > renders with custom className 1`] = `
<div>
  <div
    class="card w-80 h-80 flex flex-col items-start"
  >
    <div
      class="card__image-container w-full h-48 mb-8"
    >
      <img
        alt="John Doe"
        class="card__image w-full h-full object-cover rounded-2xl"
        src="/images/team/member.jpg"
      />
    </div>
    <div
      class="card__content flex flex-col gap-8"
    >
      <div
        class="card__text space-y-1"
      >
        <h2
          class="card__title text-2xl font-semibold text-bluedot-darker"
        >
          John Doe
        </h2>
        <p
          class="card__subtitle text-base text-black"
        >
          Developer
        </p>
      </div>
      <a
        class="cta-button flex items-center justify-center rounded-lg transition-all duration-200 font-semibold text-base cta-button--secondary bg-transparent text-bluedot-normal border border-bluedot-normal hover:bg-bluedot-lighter px-4 py-2"
        href="https://linkedin.com/in/johndoe"
        rel="noopener noreferrer"
        target="_blank"
      >
        <span
          class="cta-button__text"
        >
          LinkedIn
        </span>
      </a>
    </div>
  </div>
</div>
`;

exports[`Card > renders with custom image dimensions 1`] = `
<div>
  <div
    class="card w-80 h-80 flex flex-col items-start"
  >
    <div
      class="card__image-container w-full h-48 mb-8"
    >
      <img
        alt="John Doe"
        class="card__image w-full h-full object-cover rounded-2xl"
        src="/images/team/custom-size.jpg"
      />
    </div>
    <div
      class="card__content flex flex-col gap-8"
    >
      <div
        class="card__text space-y-1"
      >
        <h2
          class="card__title text-2xl font-semibold text-bluedot-darker"
        >
          John Doe
        </h2>
        <p
          class="card__subtitle text-base text-black"
        >
          Developer
        </p>
      </div>
      <a
        class="cta-button flex items-center justify-center rounded-lg transition-all duration-200 font-semibold text-base cta-button--secondary bg-transparent text-bluedot-normal border border-bluedot-normal hover:bg-bluedot-lighter px-4 py-2"
        href="https://linkedin.com/in/johndoe"
        rel="noopener noreferrer"
        target="_blank"
      >
        <span
          class="cta-button__text"
        >
          LinkedIn
        </span>
      </a>
    </div>
  </div>
</div>
`;
